### ISSUE TITLE
Stop using the HHVM test runner

### ISSUE THREAD

**Issue by fredemmott** [hhvm/user-documentation#863]
Future work: blocked by having a good parallelism API. Likely going to be something convenient around xbox.

We have a mostly-compatible implementation of .expectf, .expectregex already for .typechecker.expectf: we no longer use `run.php --typechecker`: we can assume that 'No errors' tests have been verified by the repo-level `hh_client` call - so we can skip them for performance.

We use it for runtime tests still as we really really want the parallelism it offers, as we can't take a similar shortcut.

**Comment by fredemmott** [comment]
It should remain compatible (or 'close enough') so that people working on features can contribute documentation examples in a format they're familiar with.

**Comment by fredemmott** [comment]
Actually a big chain of blockers here:
1. run tests in parallel
2. ... add a zero-config way to do parallelism in hack
3. decide on an entrypoint mechanism for that
4. .... but internal teams don't want a new entrypoint mechanism that depends on PHPisms, so add a $GLOBALS-free entrypoint mechanism
5. ... which also includes the input/output handles, so get HSL IO builtin
